src/acl/sesman.hpp

```diff
@@ -140,10 +140,12 @@ struct SesmanInterface
 
     bool is_capture_necessary()
     {
-        return (!ini.get<cfg::globals::is_rec>()
-            && bool(ini.get<cfg::video::disable_keyboard_log>() & KeyboardLogFlags::syslog)
+        bool res = static_cast<bool>(ini.get<cfg::video::disable_keyboard_log>() & KeyboardLogFlags::syslog)
             && !::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_kill>().c_str())
-            && !::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_notify>().c_str()));
+            && !::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_notify>().c_str());
+
+        return ini.get<cfg::video::allow_rt_without_recording>() ?
+            res : (!ini.get<cfg::globals::is_rec>() && res);
```

Comment:
This will not work if keyboard log and pattern (kbd or ocr) detection are disabled.
I think this can be as simple as:
```diff
-        return (!ini.get<cfg::globals::is_rec>()
+        bool res = (!ini.get<cfg::globals::is_rec>()
            && bool(ini.get<cfg::video::disable_keyboard_log>() & KeyboardLogFlags::syslog)
            && !::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_kill>().c_str())
            && !::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_notify>().c_str()));
+        return ini.get<cfg::video::allow_rt_without_recording>() || res;
```

Reply:
Does should be rather : "return ini.get<cfg::video::allow_rt_without_recording>() ? false : res" ?

Because when I execute in record mode your code suggestion and I put allow_rt_without_recording = true, the code where SesmanInterface::is_capture_necessary() is called in Front::can_be_start_capture() will return true and goes in the condition and prevent to instantiate Capture class :

if (sesman.is_capture_necessary())
{
            LOG(LOG_INFO, "Front::can_be_start_capture: Capture is not necessar\
y");
            return false;
}

Furthermore, the log in condition is weird because when capture is necessary, we log that capture is not necessary...

Reply:
You are right, the name of the function is misleading.
```
        return (ini.get<cfg::video::allow_rt_without_recording>()
            || ini.get<cfg::globals::is_rec>()
            || !bool(ini.get<cfg::video::disable_keyboard_log>() & KeyboardLogFlags::syslog)
            || ::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_kill>().c_str())
            || ::contains_kbd_or_ocr_pattern(ini.get<cfg::context::pattern_notify>().c_str()));
```
and change the call in front.hpp
```diff
-        if (sesman.is_capture_necessary())
+        if (!sesman.is_capture_necessary())
```